packages/invoices/src/OutboundShipment/DetailView/reducer.ts

```diff
@@ -64,17 +70,65 @@ export const OutboundAction = {
     type: ActionType.UpdateNumberOfPacks,
     payload: { rowKey, numberOfPacks },
   }),
-  onSortBy: (column: Column<OutboundShipmentRow>): OutboundShipmentAction => ({
+  onSortBy: (
+    column: Column<OutboundShipmentSummaryItem>
+  ): OutboundShipmentAction => ({
     type: ActionType.SortBy,
     payload: { column },
   }),
 };
 
 export interface OutboundShipmentStateShape {
   draft: OutboundShipment;
-  sortBy: SortBy<OutboundShipmentRow>;
+  sortBy: SortBy<OutboundShipmentSummaryItem>;
 }
 
+type InvoiceLinesByItemId = Record<string, OutboundShipmentRow[]>;
+
+const createItem = (
+  itemId: string,
+  batches: OutboundShipmentRow[]
+): OutboundShipmentSummaryItem => {
+  const item: OutboundShipmentSummaryItem = {
+    id: itemId,
+    itemId: itemId,
+    itemName: ifTheSameElseDefault(batches, 'itemName', ''),
+    itemCode: ifTheSameElseDefault(batches, 'itemCode', ''),
+    itemUnit: ifTheSameElseDefault(batches, 'itemUnit', ''),
+    batches,
+    unitQuantity: batches.reduce(getUnitQuantity, 0),
+    numberOfPacks: batches.reduce(getSumOfKeyReducer('numberOfPacks'), 0),
+    locationDescription: ifTheSameElseDefault(
+      batches,
+      'locationDescription',
+      undefined
+    ),
+
+    batch: ifTheSameElseDefault(batches, 'batch', '[multiple]'),
+    // TODO: Likely should just be a string.
+    sellPrice: ifTheSameElseDefault(batches, 'sellPricePerPack', undefined),
+    // TODO: Likely should just be a string.
+    packSize: ifTheSameElseDefault(batches, 'packSize', undefined),
+  };
+
+  return item;
+};
+
+const createItems = (lines: OutboundShipmentRow[]) => {
+  const byItem: InvoiceLinesByItemId = lines.reduce((acc, line) => {
+    const itemId = line.itemId;
+    const lines = acc[itemId]
+      ? [...(acc[itemId] as OutboundShipmentRow[]), line]
+      : [line];
+    acc[itemId] = lines;
+    return acc;
+  }, {} as InvoiceLinesByItemId);
+
+  return Object.keys(byItem).map(itemId =>
+    createItem(itemId, byItem[itemId] as OutboundShipmentRow[])
+  );
+};
+
```

Comment:
essentially this is the nuts and bolts...

rather than having the table render `OutboundShipmentRow` from `draft.lines`, have this array mapped to an array of `OutboundShipmentItem`s, which is essentially `{itemId: string, batches: OutboundShipmentRow[]}`

Then, the table can render each item. Each row is based an item, and the expando can take `rowData.batches` and then when we click to edit, we can set the `editting line` to be this item, which passes all the invoice lines for that item.

changes a bit of stuff, so wanted to try and get feedback before I went on with it..

Reply:
If we need to toggle between per batch and summary lines, then we can switch between rendering an array of `OutboundShipmentItem` or `OutboundShipmentRow`

Reply:
Though, I would want to avoid having duplicated copies in state, so maybe using ids to reference objects, or something